### ISSUE TITLE
Auto classes search space improvement

### ISSUE THREAD

**Issue by candalfigomoro** [Nixtla/neuralforecast#553]
I think the default search spaces of the Auto classes could be improved. I will focus on `AutoNHITS`, but the points should apply to Auto* classes in general.

This is the default AutoNHITS search space:
```
    default_config = {
        "input_size_multiplier": [1, 2, 3, 4, 5],
        "h": None,
        "n_pool_kernel_size": tune.choice(
            [3 * [1], 3 * [2], 3 * [4], [8, 4, 1], [16, 8, 1]]
        ),
        "n_freq_downsample": tune.choice(
            [
                [168, 24, 1],
                [24, 12, 1],
                [180, 60, 1],
                [60, 8, 1],
                [40, 20, 1],
                [1, 1, 1],
            ]
        ),
        "learning_rate": tune.loguniform(1e-4, 1e-1),
        "scaler_type": tune.choice([None, "robust", "standard"]),
        "max_steps": tune.choice([500, 1000]),
        "batch_size": tune.choice([32, 64, 128, 256]),
        "windows_batch_size": tune.choice([128, 256, 512, 1024]),
        "loss": None,
        "random_seed": tune.randint(1, 20),
    }
```

1) The NHITS default parameters are not tried in AutoNHITS, for example the default pool kernel size parameter in the NHITS class is `n_pool_kernel_size: list = [2, 2, 1],` (see https://github.com/Nixtla/neuralforecast/blob/main/neuralforecast/models/nhits.py#L236), but in the AutoNHITS search space the [2, 2, 1] combination is not tried. It is a bit weird to have a default value that is not even considered during hyperparameter optimization. If it is a good default value, it should probably be included in the search. The same is true for the `n_freq_downsample` parameter.

2) `"input_size_multiplier": [1, 2, 3, 4, 5]` uses discrete 1, 2, 3, 4, 5 values. Perhaps it would be better to take a random integer from 1 to 5 (included) so that we can take advantage of their ordinality?

3) `"max_steps": tune.choice([500, 1000])`. Perhaps a more granular space would be better? For example, a random integer value from 100 to 1000 in steps of 100? We could take advantage of the ordinality of the values here too.

4) `"batch_size": tune.choice([32, 64, 128, 256])`. Perhaps 2^(random integer number from 5 to 8)? We could take advantage of the ordinality of the values here too, since 32 is closer to 64 than to 256.

5) `"windows_batch_size": tune.choice([128, 256, 512, 1024])`. Perhaps 2^(random integer number from 7 to 10)?

6) `"random_seed": tune.randint(1, 20)`. Perhaps the seed should simply be fixed and not optimized? In my experience of 10 years in ML, trying to optimize the seed has always been considered weird and not exactly a best practice.

Thanks!

**Comment by kdgutier** [comment]
Hey @candalfigomoro, 

Thanks for the suggestions on the definition of the AutoNHITS space.
We incorporated most of the recommendations, in particular the ones that help the search engine to leverage the validation signal smoothness.
